File: CodeForces/x.py

#def findCoin(mean, coins):
    

if __name__ == "__main__":
    tests = int(input())
    for i in range(tests):
        n = int(input())
        coins = input().split()
        mean = (sum(int(i) for i in coins)/n)
        if float(mean) > int(mean):
            print("Impossible")
    
        if coins.count(coins[0])==len(coins):
            print(1)
            break

        for i in coins:
            if int(i)==int(mean):
                print(int(mean))

File: CodeForces/x.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;

bool check(const int array[], int n)
{   
    for (int i = 0; i < n - 1; i++)      
    {         
        if (array[i] != array[i + 1])
            return true;
    }
    return false;
}


int main(){
    int tests;
    cin >> tests >> endl;
    for (int i=0; i<tests; i++){
        int n, arr[n], sum = 0;
        for (int j=0; j<n; j++){
            cin >> arr*;
            sum+=arr[j];
        }
        float mean = sum/n;
        int mean1 = sum/n;

        if (mean > mean1){
            cout << "Impossible";
        }

        if (check(arr, n)==0){
            cout << 1;
        }else{
            cout << mean1;
        }
    }
}